src/feedback/commands/submitFeedbackListener.ts

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { _Blob } from 'aws-sdk/clients/lambda'
+import { getLogger, Logger } from '../../shared/logger'
+import { TelemetryService } from '../../shared/telemetry/telemetryService'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+
+export interface FeedbackMessage {
+    command: string
+    comment?: string
+    sentiment?: string
+}
+
+export interface FeedbackPanel {
+    postMessage(message: any): Thenable<boolean>
+    dispose(): any
+}
+
+export interface Window {
+    showInformationMessage(message: string): Thenable<string | undefined>
+}
+
+export function submitFeedbackListener(panel: FeedbackPanel, window: Window, telemetry: TelemetryService) {
+    const logger: Logger = getLogger()
+
+    return async (message: FeedbackMessage) => {
+        switch (message.command) {
+            case 'submitFeedback':
+                logger.info(`Submitting ${message.sentiment} feedback`)
+
+                try {
+                    await telemetry.postFeedback({
+                        comment: message.comment || '',
```

Comment:
Do we need to sanitize the comment?

Reply:
I thought about this. I don't *think* it's an issue since we never try to interpret /execute anything related to the string AFAIK.